kratos/utilities/beam_math_utilities.hpp

```diff
@@ -539,46 +661,47 @@ template<class TDataType> class BeamMathUtils
   //*****************************************************************************
 
   /**
-   * Deffault expression for GID local beam axes:: E3 is considered the local beam axial direction
-   * @param rLocalZ: Local Beam axis director vector E3 (input parameter) (output parameter)
-   * @param rLocalX: Local Beam axis director vector E1 (output parameter)
+   * Deffault expression for GID local beam axes:: E1 is considered the local beam axial direction
+   * @param rLocalX: Local Beam axis director vector E1 (input parameter) (output parameter)
    * @param rLocalY: Local Beam axis director vector E2 (output parameter)
+   * @param rLocalZ: Local Beam axis director vector E3 (output parameter)
    */
-  static inline  void CalculateLocalAxesVectors(VectorType& rLocalZ, VectorType& rLocalX, VectorType& rLocalY)
+  template<class TVector3>
+  static inline  void CalculateLocalAxesVectors(TVector3& rLocalX, TVector3& rLocalY, TVector3& rLocalZ)
   {
 
     KRATOS_TRY
 
-    VectorType GlobalY = ZeroVector(3);
+    TVector3 GlobalY = ZeroVector(3);
     GlobalY[1]=1.0;
 
-    VectorType GlobalZ = ZeroVector(3);
+    TVector3 GlobalZ = ZeroVector(3);
     GlobalZ[2]=1.0;
 
-    // local z-axis (e3_local) is the beam axis
-    double VectorNorm = MathUtilsType::Norm(rLocalZ);
+    // local x-axis (e1_local) is the beam axis
+    double VectorNorm = MathUtilsType::Norm(rLocalX);
     if( VectorNorm != 0)
-      rLocalZ /= VectorNorm;
+      rLocalX /= VectorNorm;
     
-    // local x-axis (e1_local)  
+    // local y-axis (e2_local)  
     double tolerance = 1.0/64.0;
-    if(fabs(rLocalZ[0])< tolerance && fabs(rLocalZ[1])< tolerance){
-      rLocalX = MathUtilsType::CrossProduct(GlobalY, rLocalZ);
+    if(fabs(rLocalX[0])< tolerance && fabs(rLocalX[1])< tolerance){
+      rLocalY = MathUtilsType::CrossProduct(GlobalY, rLocalX);
```

Comment:
I will check it

Reply:
Yes I see a little bit of disorder in the CrossProduct functions of math utils. This is why I am using Vector as a type not array_1d. I think this is correct:

    static inline Vector CrossProduct( Vector& a,   Vector& b )
    {
        Vector c(3);
        c[0] = a[1]*b[2] - a[2]*b[1];
        c[1] = a[2]*b[0] - a[0]*b[2];
        c[2] = a[0]*b[1] - a[1]*b[0];
        return c;
    }
It is not the most efficient way but ... For me it will be ok to change it when the CrossProduct and the UnitCrossProduct would be correctly implemented in the MathUtils. For a template vector and having both options two or three parameters. I understand that with three parameters it is more efficient, but I can detect the mess it is to know if is the first or the last vector the one it is returned.